### DISCUSSION TITLE
New Config Help

### DISCUSSION THREAD

**Discussion by jeffwchen** [devcompl/TTController#143] [Configuration] [answered]
Hi,

I have a new build with the Floe Riing 360 AIO (with 3 fans) along with 3 Riing Trio case fans.  Here is my report (I believe the AIO/radiator fans were IDed as Riing Quad).  I'd like to set the initial effect as Aurora and I would also like to know how to turn off the RGB lighting effects when desired.  Thanks in advance for your help!

```
Info
-------------------------------
»
»        OS: Windows 10 Home 2009 [19041.1.amd64fre.vb_release.191206-1406]
»        Build: 0.11.0-beta+Branch.master.Sha.42768ba987b22e18ff8cdd114f754e5f7ce16d72
»
-------------------------------
Applications
-------------------------------
»
»
-------------------------------
HID
-------------------------------
»
»  [0x264a, 0x2135]: \\?\hid#vid_264a&pid_2135#8&1e96a542&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
»  [0x264a, 0x2261]: \\?\hid#vid_264a&pid_2261#9&48bd5d6&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
»
-------------------------------
Controllers
-------------------------------
»
»  Name: Riing Quad
»        VendorId: 9802
»        ProductId: 8801
»        Version: 3.0.3
»        Ports:
»                1:
»                        Data: [PortId: 1, Speed: 50, Rpm: 894, Unknown: 255]
»                        Identifier: [9802, 8801, 1]
»                2:
»                        Data: [PortId: 2, Speed: 50, Rpm: 923, Unknown: 255]
»                        Identifier: [9802, 8801, 2]
»                3:
»                        Data: [PortId: 3, Speed: 50, Rpm: 907, Unknown: 255]
»                        Identifier: [9802, 8801, 3]
»                4:
»                        Data: [PortId: 4, Speed: 100, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8801, 4]
»                5:
»                        Data: [PortId: 5, Speed: 60, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8801, 5]
»        Available effect types: PerLed
»  Name: Riing Trio
»        VendorId: 9802
»        ProductId: 8501
»        Version: 3.1.1
»        Ports:
»                1:
»                        Data: [PortId: 1, Speed: 50, Rpm: 812, Unknown: 255]
»                        Identifier: [9802, 8501, 1]
»                2:
»                        Data: [PortId: 2, Speed: 50, Rpm: 795, Unknown: 255]
»                        Identifier: [9802, 8501, 2]
»                3:
»                        Data: [PortId: 3, Speed: 50, Rpm: 810, Unknown: 255]
»                        Identifier: [9802, 8501, 3]
»                4:
»                        Data: [PortId: 4, Speed: 60, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8501, 4]
»                5:
»                        Data: [PortId: 5, Speed: 60, Rpm: 0, Unknown: 255]
»                        Identifier: [9802, 8501, 5]
»        Available effect types: PerLed
»
-------------------------------
Sensors
-------------------------------
»
»  Nuvoton NCT6687D:
»        Temperature:
»                CPU (/lpc/nct6687d/temperature/0): 51.0 °C
»                System (/lpc/nct6687d/temperature/1): 36.0 °C
»                VRM MOS (/lpc/nct6687d/temperature/2): 38.0 °C
»                PCH (/lpc/nct6687d/temperature/3): 36.5 °C
»                CPU Socket (/lpc/nct6687d/temperature/4): 35.0 °C
»                PCIe x1 (/lpc/nct6687d/temperature/5): 35.5 °C
»                M2_1 (/lpc/nct6687d/temperature/6): 0.0 °C
»
»  AMD Ryzen 9 5900X:
»        Temperature:
»                Core (Tctl/Tdie) (/amdcpu/0/temperature/2): 51.9 °C
»                CCD1 (Tdie) (/amdcpu/0/temperature/3): 46.8 °C
»                CCD2 (Tdie) (/amdcpu/0/temperature/4): 45.5 °C
»                CCDs Max (Tdie) (/amdcpu/0/temperature/5): 46.8 °C
»                CCDs Average (Tdie) (/amdcpu/0/temperature/6): 46.1 °C
»
»  NVIDIA GeForce RTX 3080:
»        Temperature:
»                GPU Core (/gpu-nvidia/0/temperature/0): 49.0 °C
»
»  WDS100T3X0C-00SJG0:
»        Temperature:
»                Temperature (/nvme/0/temperature/0): 45.0 °C
```

**Comment by MoshiMoshi0** [comment, marked as answer]
It would have been faster for you to just search for configs with aurora effect in other discussions and copying it.

> I would also like to know how to turn off the RGB lighting effects when desired

Its possible to control the service but there is no gui for that, it has to be yet to be made by someone or me. So you would have to send data manually. You can change effects based on time of day if that's what you need.

You need to change `"CurvePoints"` for your desired pwm fan curve. I assumed `[9802, 8801, 4]` is the pump.
```json
{
  "Profiles": [
    {
      "Name": "Default",
      "Ports": [
        [9802, 8801, 1],
        [9802, 8801, 2],
        [9802, 8801, 3],
        [9802, 8501, 1],
        [9802, 8501, 2],
        [9802, 8501, 3],
        [9802, 8801, 4]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController",
          "Config": {
            "CurvePoints": [
              [50, 20],
              [70, 30],
              [80, 50],
              [85, 75],
              [90, 100]
            ],
            "Sensors": ["/amdcpu/0/temperature/2"],
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ],
      "Effects": [
        {
          "Type": "AuroraEffect",
          "Config": {
            "Step": 0.003,
            "Length": 64,
            "Mirror": false,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 8501, 1],
        [9802, 8501, 2],
        [9802, 8501, 3]
      ],
      "Config": {
        "DeviceType": "RiingTrio"
      }
    },
    {
      "Ports": [
        [9802, 8801, 4]
      ],
      "Config": {
        "DeviceType": "FloeRiing"
      }
    }
  ]
}
```

**Reply by jeffwchen** [reply]
Thanks for the reply, and yes [9802, 8801, 4] is the pump. BTW is there any need to list the following under the PortConfigs section (the fans on the radiator managed by the Quad controller):

        [9802, 8801, 1],
        [9802, 8801, 2],
        [9802, 8801, 3]

Is there a config profile that would leave the RGB lights off all the time if I decided I just didn't want to run any effects at all, but of course continue having the fans functioning? So not necessarily a GUI switch, but just swapping in a non-RGB profile basically. Thanks for your help thus far!

**Reply by MoshiMoshi0** [reply]
> BTW is there any need to list the following under the PortConfigs section (the fans on the radiator managed by the Quad controller)

Yes, otherwise the service wont know what is connected to those ports and will default to 12 leds.

> Is there a config profile that would leave the RGB lights off all the time if I decided I just didn't want to run any effects at all

Just change the effect to [StaticColorEffect](https://moshimoshi0.github.io/TTController/plugins/effects/staticcoloreffect):
```json
        {
          "Type": "StaticColorEffect",
          "Config": {
            "Color": {
              "Full": [0, 0, 0]
            },
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
```


Also, I forgot to add it to the config but since you have a Riing Quad controller you need to add a `"PortConfig"` with [IgnoreColorCache](https://moshimoshi0.github.io/TTController/config/port#ignorecolorcache) for its ports:
```json
    {
      "Ports": [
        [9802, 8801, 1],
        [9802, 8801, 2],
        [9802, 8801, 3],
        [9802, 8801, 4]
      ],
      "Config": {
        "IgnoreColorCache": true
      }
    }
```

**Reply by jeffwchen** [reply]
Thanks for the response. I was able to get things working. I just had a couple more quick questions:

1) Under PortConfigs would there be any need to separate out the FloeRiing pump and the radiator fans, which I believe are Riing Plus fans?  I'm just wondering whether those parts of the AIO may have a different number of addressable LEDs even though they came together.

2) After searching some of the other discussions I noticed some of the profiles have a section that sets a Sensor critical temperature.  Is that needed at all?

Thanks so much, and I'm happy to make a donation for this project.  I wasn't able to use TT RGB Plus, as it kept crashing and freezing my system, so it's really great to be able to have something that actually works!

**Reply by MoshiMoshi0** [reply]
> Under PortConfigs would there be any need to separate out the FloeRiing pump and the radiator fans, which I believe are Riing Plus fans? I'm just wondering whether those parts of the AIO may have a different number of addressable LEDs even though they came together.

The default is 12 leds which Riing Plus radiator fans have, so they are not specified in port configs, pump has only 6. 

> After searching some of the other discussions I noticed some of the profiles have a section that sets a Sensor critical temperature. Is that needed at all?

Not technically needed but you can add it as a safety, if a sensor reaches the critical temp the device speed is forced to 100 on all ports. Can happen for example when you have a very flat/quiet pwm curve and you suddenly start using 100% of your cpu for a some time, it can take a bit of time for the fans to ramp up depending on settings.

Thanks!

**Comment by jeffwchen** [comment]
Ok thanks for the additional info. FYI I know this seemed to mainly be an
issue with the Quad controller based on the other posts, but I found I had
to also add IgnoreColorCache to the RiingTrio PortConfigs section too to
prevent it from reverting to the default RGB pattern.
